### PR TITLE
In VB we shouldn't deprioritize Add-Import fixes if they involve a case-correction

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://github.com/dotnet/roslyn/issues/11241

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Tagging @dotnet/roslyn-ide

**Comment by CyrusNajmabadi** [comment]
retest vsi please

File: src/EditorFeatures/VisualBasicTest/Diagnostics/AddImport/AddImportTests.vb [hunk 1]
```diff
@@ -121,7 +149,7 @@ NewLines("Class SomeClass \n Sub Foo(Of [|SomeClass|])(x As SomeClass) \n End Su
 NewLines("Class SomeClass \n Sub Foo(Of SomeClass)(x As [|SomeClass|]) \n End Sub \n End Class \n Namespace SomeNamespace \n Class SomeClass \n End Class \n End Namespace"))
         End Function
 
-        <WorkItem(540543, "http://vstfdevdiv:8080/DevDiv2/DevDiv/_workitems/edit/540543")>
+        <WorkItem(540543, "http//vstfdevdiv:8080/DevDiv2/DevDiv/_workitems/edit/540543")>
```
Comment (Pilchie) [inline src/EditorFeatures/VisualBasicTest/Diagnostics/AddImport/AddImportTests.vb]: Why did you do this to the whole file?

**Comment by Pilchie** [comment]
LGTM except all the `WorkItem` attribute changes.

File: src/EditorFeatures/VisualBasicTest/Diagnostics/AddImport/AddImportTests.vb (hunk 1 above)
Comment (CyrusNajmabadi) [inline src/EditorFeatures/VisualBasicTest/Diagnostics/AddImport/AddImportTests.vb]: I didn't...  that's super super weird.  Some other system (hopefully not in roslyn) must have screwed that up!

**Comment by jmarolf** [comment]
:shipit:

File: src/Features/Core/Portable/CodeFixes/AddImport/AbstractAddImportCodeFixProvider.SymbolResult.cs [hunk 2]
```diff
@@ -38,12 +39,19 @@ public SearchResult(string desiredName, TSimpleNameSyntax nameNode, IReadOnlyLis
...
             {
                 return !string.IsNullOrEmpty(this.DesiredName) &&
                     this.NameNode != null &&
                     this.NameNode.GetFirstToken().ValueText != this.DesiredName;
             }
+
+            public bool DesiredNameDiffersFromSourceNameOnlyByCase()
+            {
+                Debug.Assert(DesiredNameDiffersFromSourceName());
+                return StringComparer.OrdinalIgnoreCase.Equals(
```
Comment (jasonmalinowski) [inline src/Features/Core/Portable/CodeFixes/AddImport/AbstractAddImportCodeFixProvider.SymbolResult.cs]: There's a `CaseInsensitiveComparison` public API that implements the actual logic the compiler uses for identifier equality.

**Comment by jasonmalinowski** [comment]
One comment to use the correct string comparison API, otherwise :+1:.

**Pull request merged**